### ISSUE TITLE
Confusing return type in many solvers

### ISSUE THREAD

**Issue by fabianp** [google/jaxopt#251]
For example, in LBFGS

![image](https://user-images.githubusercontent.com/277639/174670346-e4b0d50f-df11-4e26-8a64-71dc923587f6.png)

Reading the Return type, it would seem like this returns an object of type OptStep, while in reality, it returns a tuple, of which the second argument is of type OptStep

**Comment by mblondel** [comment]
Nope, that's correct https://github.com/google/jaxopt/blob/main/jaxopt/_src/base.py#L48

**Comment by fabianp** [comment]
oh, got it. It would be nice though if the OptStep was clickable. I don't think it's documented (tried pasting it in the search bar and didn't get anywhere). I'll add another issue for this

**Comment by mblondel** [comment]
+1 on exporting and making it clickable
